File: WPF/MVVM/Converters/BooleanToVisibilityConverter.vb
﻿Imports System.Globalization
Imports System.Windows.Data

Namespace Common.MVVM.Converters

    Public Class BooleanToVisibilityConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            If CType(Parameter, String) = "~" Xor CType(Value, Boolean) Then
                Return Visibility.Visible
            Else
                Return Visibility.Collapsed
            End If
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return (CType(Parameter, String) = "~") Xor CType(Value, Visibility) = Visibility.Visible
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/BooleanToVisibilityConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace Ks.Common.MVVM.Converters
{
    public class BooleanToVisibilityConverter : IValueConverter
    {
        public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            if ((System.Convert.ToString(Parameter) == "~") ^ System.Convert.ToBoolean(Value))
            {
                return Visibility.Visible;
            }
            else
            {
                return Visibility.Collapsed;
            }
        }

        public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            return (System.Convert.ToString(Parameter) == "~") ^ ((Visibility) Value == Visibility.Visible);
        }
    }
}
